src/d/parser/expression.d

```diff
@@ -1047,3 +1050,36 @@ StringLiteral parseStringLiteral(ref TokenRange trange) {
 	
 	return new StringLiteral(location, name.toString(trange.context));
 }
+
+/**
+ * Parse floating point literals
+ */
+FloatLiteral parseFloatLiteral(ref TokenRange trange) {
+	const location = trange.front.location;
+	auto litString = trange.front.toString(trange.context);
+
+	trange.match(TokenType.FloatLiteral);
+	import d.common.builtintype : BuiltinType;
+
+	//Look for a suffix
+	if (litString.length > 1)
+	{
+		switch (litString[$-1])
+		{
+			// https://dlang.org/spec/lex.html#FloatSuffix
+			import std.conv : to;
+			case 'f':
+			case 'F':
+				const float f = litString[0..$-1].to!float;
+				return new FloatLiteral(location, f, BuiltinType.Float);
+			case 'L':
+				import source.exception;
+				throw new CompileException(location, "SDC does not support real literals yet");
```

Comment:
There is no reason not to generate a node in the AST at this level, the only difference should be the type. This should fail later on in the pipeline :P